README.md

```diff
@@ -96,6 +104,46 @@ meta = {
 */
 ```
 
+## Using with Next.js `getServerSideProps`, `getInitialProps`, and `getStaticProps`
+
+The `getServerSideProps`, `getInitialProps`, and `getStaticProps` data hooks provided by Next.js do not allow you to transmit Javascript objects like Dates. It will error unless you convert Dates to strings, etc.
+
+Thankfully, Superjson is a perfect tool to bypass that limitation!
+
+```js
+import { useMemo } from 'react';
+import superjson from 'superjson';
+
+export const getServerSideProps = async () => {
+  const products = [{ name: 'Hat', publishedAt: new Date(0) }];
+
+  const dataString = superjson.stringify(products);
+
+  return {
+    props: {
+      dataString,
+    },
+  };
+};
+
+export default function Page({ dataString }) {
+  const products = useMemo(() => superjson.parse(dataString), [dataString]);
```

Comment:
When using this more often, a custom hook would probably be a good thing:

```ts
function useSuperJSON<T>(string: string): T {
  return useMemo(() => superjson.parse<T>(string), [string]);
}
```

What do you think, should we just propose the custom hook in this demo? Or are they complicated to beginners?

Reply:
Why stringify/parse rather than serialize/deserialize here?

Reply:
stringify/parse because you don't need direct access to `json` and `meta` in this case.

The hook is a good idea, but let's just merge this as is since we have the babel plugin coming which is even better.